### PR TITLE
Prefix netex as namespace of DistributionGroupRef

### LINKED ISSUE #448: DistributionGroupRef is not prefixing an explicit namespace
This uncovered an issue in my code. But I guess it is actually a bug in the schema.

https://github.com/NeTEx-CEN/NeTEx/blob/master/xsd/NeTEx_publication.xsd#L879

### PR DESCRIPTION
**Pull request by skinkie**
Fix #448

### CODE REVIEW

**Comment by skinkie** [comment]
@ue71603 can you take a peak how to make the examples working?

**Comment by ue71603** [comment]
@skinkie  I think the part iabout the DistributionGroupRef is not needed as DistributionGroupRef is a restriction of the GeneralGroupOfEntitiesRefStructure
`				</xsd:element>
				<xsd:element name="DistributionGroupRef" minOccurs="0">
					<xsd:annotation>
						<xsd:documentation>Prefined Group of specified points to which  distribution is restricted, if any.</xsd:documentation>
					</xsd:annotation>
					<xsd:complexType>
						<xsd:simpleContent>
							<xsd:restriction base="GeneralGroupOfEntitiesRefStructure">
								<xsd:attribute name="nameOfMemberClass" `

What do you think?
The other occasions with a | are  only related by SubstitutionGroup

**Comment by skinkie** [comment]
I think we should have at least one example that actually _is_ using a DistributionGroupRef. It is obviously that since we started fixing the constraints these issues popped up.

**Comment by ue71603** [comment]
The failed ones used it:

`<GroupOfPlaces version="1.0" id="frst:travel_shops">
`

`<DistributionGroupRef version="1.0" ref="frst:Travel_shops"/>
`

**Comment by ue71603** [comment]
what do you think about the latest commit?

**Comment by ue71603** [comment]
...at least the DistributionGroupRef is fixed...

File: xsd/NeTEx_publication.xsd [hunk 1]
```diff
@@ -49,8 +49,7 @@
 						ParkingAreaCapacityAssignment, MonitoredVehicleSharingParkingBay, ParkingBayStatus.
 						VehicleReleaseEquipment, RefuellingEquipment.
 						TaxiServicePlaceAssignment, VehiclePoolingPlaceAssignment, VehicleSharingPlaceAssignment.
-						SingleJourneyPath, VehicleMeetingPointInPath, SingleJourney, GroupOfSingleJourneys.
-						 VehiclePoolerProfile (USerProfile)
+						SingleJourneyPath, VehicleMeetingPoi
```
Comment (skinkie) [inline xsd/NeTEx_publication.xsd:52]: Something fishie happened here.

Comment (skinkie) [reply xsd/NeTEx_publication.xsd:52]:
```suggestion
						SingleJourneyPath, VehicleMeetingPointInPath, SingleJourney, GroupOfSingleJourneys.
						VehiclePoolerProfile (USerProfile)
```

File: xsd/NeTEx_publication.xsd [hunk 2]
```diff
@@ -50,7 +50,7 @@
 						VehicleReleaseEquipment, RefuellingEquipment.
 						TaxiServicePlaceAssignment, VehiclePoolingPlaceAssignment, VehicleSharingPlaceAssignment.
 						SingleJourneyPath, VehicleMeetingPointInPath, SingleJourney, GroupOfSingleJourneys.
-						 VehiclePoolerProfile (USerProfile)
+						VehiclePoolerProfile (USerProfile)
```
Comment (skinkie) [inline xsd/NeTEx_publication.xsd:53]:
```suggestion
						 VehiclePoolerProfile (USerProfile)
```

File: examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml [hunk 3]
```diff
@@ -5197,6 +5197,7 @@ You can't
 							<!-- TO DO -->
 							<PaymentMethods>debitCard creditCard   cash</PaymentMethods>
 							<DistributionGroupRef ref="nr:OysterTicketShops"/>
+							<!-- no version, not validated (and also not in this file) -->
```
Comment (ue71603) [inline examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5200]:
```suggestion
							<DistributionGroupRef ref="nr:OysterTicketShops" version="any"/>
```

File: examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml [hunk 4]
```diff
@@ -5196,7 +5196,8 @@ You can't
 	Buy paper tickets (cash single tickets and Day Travelcards)</Description>
 							<!-- TO DO -->
 							<PaymentMethods>debitCard creditCard   cash</PaymentMethods>
-							<DistributionGroupRef  ref="nr:OysterTicketShops"/> <!-- no version, not validated (and also not in this file) -->
+							<DistributionGroupRef ref="nr:OysterTicketShops"/>
+							<!-- no version, not validated (and also not in this file) -->
```
[the remark this answers is not in the document]
Comment (ue71603) [reply examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml]: was already that way.. tried to fix it now.

File: examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml [hunk 5]
```diff
@@ -5196,7 +5196,7 @@ You can't
 	Buy paper tickets (cash single tickets and Day Travelcards)</Description>
 							<!-- TO DO -->
 							<PaymentMethods>debitCard creditCard   cash</PaymentMethods>
-							<DistributionGroupRef ref="nr:OysterTicketShops"/>
+							<DistributionGroupRef ref="nr:OysterTicketShops" version="any"/>
```
Comment (skinkie) [inline examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5199]:
```suggestion
							<DistributionGroupRef version="any" ref="nr:OysterTicketShops" />
```

**Review by Aurige — CHANGES_REQUESTED** [review]

File: examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml [hunk 6]
```diff
@@ -5170,7 +5170,7 @@ You can only get these as paper tickets, not on an Oyster card, and they must be
 						<DistributionChannel version="any" id="tfl:Visitor_Centres">
 							<Name>Visitor centres</Name>
 							<PaymentMethods>debitCard creditCard companyCheque cash</PaymentMethods>
-							<DistributionGroupRef version="any" ref="nr:Visitor_Centres"/>
+							<DistributionGroupRef version="any" ref="tfl:Visitor_Centres"/>
```
[the remark this answers is not in the document]
Comment (ue71603) [reply examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5173]: it worked fined. There is no nr:Visitor_Centers that could be referenced. And the id are per class so it works as well. The id references the GeneralGroupOfEntities (and there is only tfl:Visitor_Centres):
`						<GeneralGroupOfEntities version="any" id="tfl:Visitor_Centres" nameOfMemberClass="AccessSpace">
							<Name>List of visitor centres</Name>
							<members>
								<AccessSpaceRef version="1" ref="tfl:Piccadilly_Circus_Visitor_Centre"/>
								<AccessSpaceRef version="1" ref="tfl:Euston_Station_Visitor_Centre"/>
								<AccessSpaceRef version="1" ref="tfl:Victoria_Station_Visitor_Centre"/>
								<AccessSpaceRef version="1" ref="tfl:Heathrow_Terminal_123_Visitor_Centre"/>
							</members>
						</GeneralGroupOfEntities>`

it does not reference to the DistributionChannel:
`<DistributionChannel version="any" id="tfl:Visitor_Centres">
							<Name>Visitor centres</Name>
							<PaymentMethods>debitCard creditCard companyCheque cash</PaymentMethods>
							<DistributionGroupRef version="any" ref="tfl:Visitor_Centres"/>
						</DistributionChannel>`

[the remark this answers is not in the document]
Comment (Aurige) [reply examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5173]: I'm lost here:  in this example _tfl:Visitor_Centres_ is the id of the _DistributionChannel_  `<DistributionChannel version="any" id="tfl:Visitor_Centres">` that has an element refering itself as a _DistributionGroup_ ! I don't get the semantinc, neither the technical side of it !

[the remark this answers is not in the document]
Comment (ue71603) [reply examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5173]: @Aurige  This is no my example. I just made it working. That the DistributionChannel and the Group used the same id text I don't know.  Who knows? @nick-knowles ?

[the remark this answers is not in the document]
Comment (Aurige) [reply examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5173]: in fact it was ok with a different namespace, but changing the namespace to "tfl" create an inconsistency !

**Comment by ue71603** [comment]
My fault, but: It was only ok, because  no "version" was defined. With a version it no longer validated. And the other GroupofEntities was not in existance. So I used the one that was there.

File: examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml (hunk 6 above)
Comment (ue71603) [inline examples/functions/fares/Netex_101.21_TfL_GeographicFares_UnitZone_MultipleProduct.xml:5173]:
```suggestion
							<DistributionGroupRef version="any" ref="nr:Visitor_Centres"/>
```

**Review by Aurige — APPROVED** [review]

**Review by trurlurl — APPROVED** [review]

**Pull request merged**